usbl_gazebo/CMakeLists.txt

```diff
@@ -1,17 +1,11 @@
 cmake_minimum_required(VERSION 3.0.2)
 project(usbl_gazebo)
 
-## Compile as C++11, supported in ROS Kinetic and newer
-# If running on focal, will be using Noetic + Gazebo11, which requires c++17
-execute_process(COMMAND lsb_release -cs
-  OUTPUT_VARIABLE RELEASE_CODENAME
-  OUTPUT_STRIP_TRAILING_WHITESPACE
-)
-if(RELEASE_CODENAME MATCHES "focal")
-  set(CMAKE_CXX_STANDARD 17)
-  set(CMAKE_CXX_STANDARD_REQUIRED ON)
+if(NOT "${CMAKE_VERSION}" VERSION_LESS "3.16")
+    set(CMAKE_CXX_STANDARD 17)
+    set(CMAKE_CXX_STANDARD_REQUIRED ON)
 else()
-  add_compile_options(-std=c++11)
+    set(LOCAL_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
```

Comment:
Why was this change necessary? It compiled on my machine without it, and the auto-generated CMakeLists.txt seems to push people towards using `add_compile_options`.

Reply:
I am open to reverting this change back to what it was. I was following a pattern we typically use for our own packages where we prefer to set compiler flags per target rather than globally. It's a pattern we have used for a long time, and I don't really know if it makes a difference to CMake anymore (10 years ago there were plenty of edge cases where it did make a difference...).

Reply:
Absent an actual problem, I'd be inclined to keep the same pattern used by the broader project.

Reply:
> I am open to reverting this change back to what it was. I was following a pattern we typically use for our own packages where we prefer to set compiler flags per target rather than globally. It's a pattern we have used for a long time, and I don't really know if it makes a difference to CMake anymore (10 years ago there were plenty of edge cases where it did make a difference...).

@tdenewiler I agree with @lauralindzey. If you could revert that line I'll do a review.

Reply:
I didn't realize you were waiting on that change. It's reverted in af2d636 and still builds and runs for me on Noetic.